sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -191,7 +187,16 @@ protected string GetRecordedOptionalVariable(string name, Action<RecordedVariabl
                 throw new InvalidOperationException("Recorded value should not be set outside the test method invocation");
             }
 
-            _recording?.SetVariable(name, sanitizedValue);
+            // If the value was populated, sanitize before recording it.
+
+            if (!string.IsNullOrEmpty(value))
+            {
+                var optionsInstance = new RecordedVariableOptions();
+                options?.Invoke(optionsInstance);
+                value = optionsInstance.Apply(value);
+            }
+
+            _recording?.SetVariable(name, value);
             return value;
```

Comment:
You are returning sanitized value here instead of an actual one. That's why tests were on fire.

Reply:
Yeah, I just found the tests which I didn't realize were in the other solution and fixed the stupid.